add pino logger

hdahlheim opened the pull request:
Our homegrown logger setup hasn't aged well. This PR introduces the pino logger to our backend.

By using the pino-http plugin we can make sure that all logs that happen in the HTTP layer are connected to a request id.
In production pino generates structured json logs that reduce the clutter in our log drain.

- [x]: added pino-http to all http requests
- [x]: inject pino into the pgboss queue

TODO: refactor console and debug calls to pino

hdahlheim commented:
@jstcki @lucianak If you both could take a look and give feedback. I don't think refactoring everything as part of this PR makes sense. The payments worker folder contains most of the "examples" on how the new logging can be used.

jstcki commented:
I agree with not refactoring everything here but changing `req._log` to `req.log` seems pretty straightforward, so maybe still worth it?

hdahlheim commented:
@jstcki @lucianak, I removed all calls to `_log()` and replaced them with the appropriate pino log call. The PR got much bigger because of it, sorry for that, but could you please review it again?

jstcki reviewed (approved):
LGTM :)

jstcki commented:
(I haven't actually checked the logs to verify that everything works as intended)

lucianak commented on packages/backend-modules/auth/lib/Users.js:
```diff
@@ -307,7 +307,7 @@ const signIn = async (
 
 const shouldAutoLogin = ({ email }) => {
-  if (AUTO_LOGIN_REGEX) {
+  if (process.env.NODE_ENV !== 'production' && AUTO_LOGIN_REGEX) {
```
💖

lucianak commented on packages/backend-modules/base/express/graphql.js:
```diff
@@ -123,14 +125,27 @@ module.exports = async (
+                    graphqlRequest: {
+                      query: request.query,
+                      variables: request.variables,
+                      errors: errors,
```
would it be possible to log the errors using logger.error also when debug is enabled? I feel like that would make filtering logs easier. But not sure if that's feasible

lucianak commented on packages/backend-modules/payments/lib/workers/StripeWebhookWorker.ts:
```diff
@@ -95,22 +107,27 @@ export class StripeWebhookWorker extends BaseWorker<WorkerArgsV1> {
-      'successfully processed stripe event %s [%s]',
-      event.id,
-      event.type,
+    this.logger.info(
```
why is this info and not debug as for the other workers?

hdahlheim replied:
To be able to see the health of stripe event processing

lucianak reviewed (approved):
looks good!
